java/ql/test/library-tests/frameworks/apache-collections/Test.java

```diff
@@ -0,0 +1,759 @@
+			out = MapUtils.putAll(null, in);
+			sink(getMapKey(out)); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;putAll;;;MapValue of ArrayElement of Argument[1];MapValue of Argument[0];value"
+			Map out = null;
+			Object[] in = (Object[])newWithArrayElement(newWithMapValue(source()));
+			MapUtils.putAll(out, in);
+			sink(getMapValue(out)); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;putAll;;;MapValue of ArrayElement of Argument[1];MapValue of ReturnValue;value"
+			Map out = null;
+			Object[] in = (Object[])newWithArrayElement(newWithMapValue(source()));
+			out = MapUtils.putAll(null, in);
+			sink(getMapValue(out)); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;safeAddToMap;;;Argument[1];MapKey of Argument[0];value"
+			Map out = null;
+			Object in = (Object)source();
+			MapUtils.safeAddToMap(out, in, null);
+			sink(getMapKey(out)); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;safeAddToMap;;;Argument[2];MapValue of Argument[0];value"
+			Map out = null;
+			Object in = (Object)source();
+			MapUtils.safeAddToMap(out, null, in);
+			sink(getMapValue(out)); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;synchronizedMap;;;Argument[0];ReturnValue;value"
+			Map out = null;
+			Map in = (Map)source();
+			out = MapUtils.synchronizedMap(in);
+			sink(out); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;synchronizedSortedMap;;;Argument[0];ReturnValue;value"
+			SortedMap out = null;
+			SortedMap in = (SortedMap)source();
+			out = MapUtils.synchronizedSortedMap(in);
+			sink(out); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;toMap;;;Argument[0];ReturnValue;value"
+			Map out = null;
+			ResourceBundle in = (ResourceBundle)source();
+			out = MapUtils.toMap(in);
+			sink(out); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;transformedMap;;;Argument[0];ReturnValue;value"
+			IterableMap out = null;
+			Map in = (Map)source();
+			out = MapUtils.transformedMap(in, null, null);
+			sink(out); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;transformedSortedMap;;;Argument[0];ReturnValue;value"
+			SortedMap out = null;
+			SortedMap in = (SortedMap)source();
+			out = MapUtils.transformedSortedMap(in, null, null);
+			sink(out); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;unmodifiableMap;;;Argument[0];ReturnValue;value"
+			Map out = null;
+			Map in = (Map)source();
+			out = MapUtils.unmodifiableMap(in);
+			sink(out); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;unmodifiableSortedMap;;;Argument[0];ReturnValue;value"
+			SortedMap out = null;
+			SortedMap in = (SortedMap)source();
+			out = MapUtils.unmodifiableSortedMap(in);
+			sink(out); // $hasValueFlow
```

Comment:
I know that this seems like a nice shortcut, but `unmodifiableSortedMap` isn't the identity function, so modelling it as such will risk flow getting pruned by type incompatibility. Please fix.